src/databricks/sql/parameters/native.py

```diff
@@ -0,0 +1,604 @@
+import datetime
+import decimal
+from enum import Enum, auto
+from typing import Optional, TYPE_CHECKING
+
+from databricks.sql.exc import NotSupportedError
+from databricks.sql.thrift_api.TCLIService.ttypes import (
+    TSparkParameter,
+    TSparkParameterValue,
+)
+
+import datetime
+import decimal
+from enum import Enum, auto
+from typing import Dict, List, Union
+
+
+
+
+class ParameterApproach(Enum):
+    INLINE = 1
+    NATIVE = 2
+    NONE = 3
+
+
+class ParameterStructure(Enum):
+    NAMED = 1
+    POSITIONAL = 2
+    NONE = 3
+
+
+class DatabricksSupportedType(Enum):
+    """Enumerate every supported Databricks SQL type shown here:
+
+    https://docs.databricks.com/en/sql/language-manual/sql-ref-datatypes.html
+    """
+
+    BIGINT = auto()
+    BINARY = auto()
+    BOOLEAN = auto()
+    DATE = auto()
+    DECIMAL = auto()
+    DOUBLE = auto()
+    FLOAT = auto()
+    INT = auto()
+    INTERVAL = auto()
+    VOID = auto()
+    SMALLINT = auto()
+    STRING = auto()
+    TIMESTAMP = auto()
+    TIMESTAMP_NTZ = auto()
+    TINYINT = auto()
+    ARRAY = auto()
+    MAP = auto()
+    STRUCT = auto()
+
+TAllowedParameterValue = Union[
```

Comment:
I would get rid of init on base.  a.) you don't want users to instantiate a DbsqlParameterBase, and b.) it forces you to break Liskov substitution principal by having the child types take a more restricted set of inputs.